### PR TITLE
ben's API changes

### PR DESCRIPTION
**Pull request by adrian-brady**
Ben's changes made to the older main reapplied to the new main (they have diverging histories due to me rebase merging my kendra search PR and then it getting squashed)

### CODE REVIEW

**Comment by adrian-brady** [comment]
Overall things look good. I couldn't get the search bar to appear when testing myself, but the page loaded correctly otherwise. I am a bit confused about to the API. HTMX's power comes from the leveraging of HTTP requests and HTML, thus it would be best to use HTTP requests and responses and HTML in the responses as well.

**Comment by bengoh815** [comment]
# Decision Choices and Organization Summary

## Organizing and Documentation

### Routes
- Divided into two categories: APIs and pages, for clearer separation of concerns.
- Organized to call the respective api and pages packages for better modularity.

## Template Management

### Challenges
  - I had difficulty managing and importing templates using block, define, and template.
  - Current structure is functional but lacks clarity on how components integrate into pages.

### Views Organization
  - Current setup:
    - Separated into pages, components, and layouts directories for clarity.
    - **Pros**: Categories of files are grouped for easier maintenance.
    - **Cons**: Individual page components are scattered across directories.
  - Alternative Setup:
    - Organize by individual page:
    - Example:
    ```plaintext
    views/
      home/
        statistics.html
      search/
        search-bar.html
      results/
        filters.html
        book.html
    ```

    - **Pros**: All components for a single page are grouped together.
    - **Cons**: Different categories (e.g., layout, components) are separated, increasing directory depth.
 
## View and Template Structure

### Approach 1: Blanket Import
  - Import all templates or HTML files in views and resolve references dynamically.
  - **Pros**: Simplifies initial setup.
  - **Cons**:
      - Harder to trace component usage.
      - May load unnecessary templates, increasing overhead.
### Approach 2: Fine-Grained Import (Current)
  - Explicitly modify template imports whenever there are changes.
  - **Pros**: 
    - Provides granular control over templates
    - Wrong changes to template are not rendered immediately and will not crash and burn.
  - **Cons**: Developers must understand the integration logic for templates and components.
  - Should look into how templates should be managed, for easier integration. Back to Blanket Import?

## Suggestions for Asset Management

- Images: Needs better naming conventions or organize by type or use case (e.g., hero, icons, thumbnails).
- AWS Models: Consider moving AWS-related models into a dedicated directory for clarity:

## Miscellaneous

Search query not handled. Just basic GET Search page.
Look into some kind of linting or formatting for better code clarity

File: routes/routes.go [hunk 1]
```diff
@@ -2,13 +2,23 @@ package routes
 
 import (
 	"github.com/labstack/echo/v4"
+	"github.com/DSSD-Madison/gmu/routes/api"
```
Comment (sarthyparty) [inline routes/routes.go:5]: in golang, we organize imports like this:

system

external

internal

i'm not going to add a comment for all the places you need to fix this so look through your files

File: routes/routes.go [hunk 2]
```diff
@@ -2,13 +2,23 @@ package routes
...
+	*/
+	e.GET("/", pages.Home)
+	e.GET("/search", pages.Search)
 
-	// Search Route
-	e.POST("/search", Search)
+	e.GET("/results", pages.Results)
+
+	/*
+		API
```
Comment (sarthyparty) [inline routes/routes.go:20]: this is not necessary, either just make a function that returns the mock data or use the actual API

File: tailwind.config.js [hunk 3]
```diff
@@ -1,7 +1,8 @@
 /** @type {import('tailwindcss').Config} */
 module.exports = {
     content: [
-        "./views/*.html"
+        "./views/*.html",
```
Comment (sarthyparty) [inline tailwind.config.js:4]: i don't think you need this

File: views/pages/home.html [hunk 4]
```diff
@@ -0,0 +1,78 @@
+<!-- Home Page -->
+{{ define "content" }}
```
Comment (sarthyparty) [inline views/pages/home.html:2]: let's call this an info page

File: views/pages/results.html [hunk 5]
```diff
@@ -0,0 +1,13 @@
+<!-- Results Page -->
+{{ define "content" }}
+<div class="relative h-[300px]">
```
Comment (sarthyparty) [inline views/pages/results.html:3]: ok try to get everything that AJ has worked on involving results to work so we don't have to redo that

File: views/pages/book.html [hunk 6]
```diff
@@ -0,0 +1,8 @@
+<!-- Book Page -->
+{{ define "book" }}
+<div class="p-6">
+  <h2 class="text-2xl font-bold">{{ .Title }}</h2>
+  <p class="text-gray-700">{{ .Description }}</p>
+  <p class="text-sm text-gray-500">{{ .Id }}</p>
```
Comment (sarthyparty) [inline views/pages/book.html:6]: What is the reason for Id?

File: CONTRIBUTING.md [hunk 7]
```diff
@@ -0,0 +1,55 @@
+# Contributing to Better Evidence Project
+
+## Project Overview
+
```
Comment (sarthyparty) [inline CONTRIBUTING.md:4]: i'm not really sure if you want to keep this, this PR is not really for working on any documentation and i dont want to givw feedback on this

File: API.md [hunk 8]
```diff
@@ -0,0 +1,148 @@
+# API Documentation
+
+
+## Filters API
+
```
Comment (sarthyparty) [inline API.md:5]: we do not need this at all. we shouldnt have an APi for filters, we're using the kendra api

File: .gitignore [hunk 9]
```diff
@@ -1,5 +1,35 @@
...
 static/css/output.css
-.env
+
+# Ignoring compiled binaries and executables
+/bin
+
+# Ignoring temporary build files
 tmp
+
+# ========================
```
Comment (sarthyparty) [inline .gitignore:13]: whats with all the comments??

**Review by sarthyparty — CHANGES_REQUESTED** [review]
ok Ben, left a few comments to help you in the right direction. this is not a complete review and once you clean it up I can give better comments and help better, right now this is just general stuff.

You've done a lot of unnecessary work and you will need to fix that before this gets approved. I do like the filters UI  it does look good

**Pull request closed without merging**